### PR TITLE
TD Supply Truck

### LINKED ISSUE #12849: game mechanic that allows TD players to come back from no money in a team game
As of now, when a players refinery is destroyed and they lack the money to construct one they are effectively out of game because TD has no mechanic that allows team members to transfer money. Red Alert has the supply trucks and my suggestion is to implement the same, if not then a similar, thing for TD. This is open to other suggestions.

### PR DESCRIPTION
**Pull request by anjew175**
Adds supply truck to TD
closes #12849

### CODE REVIEW

File: mods/cnc/rules/campaign-maprules.yaml [hunk 1]
```diff
@@ -13,6 +13,7 @@ World:
 	MapOptions:
 		ShortGameLocked: True
 		ShortGameEnabled: False
+
```
Comment (GraionDilach) [inline mods/cnc/rules/campaign-maprules.yaml]: Revert this line.

File: mods/cnc/rules/structures.yaml [hunk 2]
```diff
@@ -66,7 +66,8 @@ FACT:
 	ProvidesPrerequisite@buildingname:
 	Buildable:
 		Description: Builds structures
-
+	AcceptsDeliveredCash:
+
```
Comment (GraionDilach) [inline mods/cnc/rules/structures.yaml]: Empty spaces in line.

File: mods/cnc/rules/structures.yaml [hunk 3]
```diff
@@ -78,7 +79,8 @@ FACT.GDI:
 		ForceFaction: gdi
 	Tooltip:
 		Name: Construction Yard (GDI)
-
+	AcceptsDeliveredCash:
+
```
Comment (GraionDilach) [inline mods/cnc/rules/structures.yaml]: Same here.

File: mods/cnc/rules/structures.yaml [hunk 4]
```diff
@@ -90,7 +92,8 @@ FACT.NOD:
 		ForceFaction: nod
 	Tooltip:
 		Name: Construction Yard (Nod)
-
+	AcceptsDeliveredCash:
+
```
Comment (GraionDilach) [inline mods/cnc/rules/structures.yaml]: And here.

File: mods/cnc/rules/campaign-maprules.yaml (hunk 1 above)
Comment (GraionDilach) [inline mods/cnc/rules/campaign-maprules.yaml]: IOW there's a tab here.

File: mods/cnc/rules/structures.yaml [hunk 5]
```diff
@@ -78,7 +79,8 @@ FACT.GDI:
 		ForceFaction: gdi
 	Tooltip:
 		Name: Construction Yard (GDI)
-
+	AcceptsDeliveredCash:
```
Comment (MustaphaTR) [inline mods/cnc/rules/structures.yaml]: This and one on FACT.NOD is not required. They already Inherit that from FACT.

Actually you can just put it on ^Building under defaults.yaml and remove it on ^Defence and others that don't need it as done in RA mod. No need to specify it on each building.

**Review by MustaphaTR — CHANGES_REQUESTED** [review]

**Review by pchote — DISMISSED** [review]
Adding the supply truck fills a known gameplay hole, so the premise of this PR is fine with me.

However, in addition to the issues raised by @GraionDilach and @MustaphaTR I have two more:

* This is a low tech unit, so the build icon should be in the top row of the palette (low BuildPaletteOrder).
* The palette didn't convert very well, the artwork looks glitchy ingame:
screen shot 2017-05-06 at 10 10 10 screen shot 2017-05-06 at 10 09 59

**Comment by GraionDilach** [comment]
Is this the RA1 Truck? If yes, maybe it would make more sense using the original artwork with the RA pal instead of converting the artwork to TD.

**Comment by pchote** [comment]
I don't want to have to pull in the RA palette (yet another non-free binary file in our repo) and bloat the palette definitions just for one unit.

**Comment by GraionDilach** [comment]
Converting palette between TD and RA requires 2-pass methods (!) which I presume is above @anjew175 's level (yes, that's required simply because of the different position of the remap gradient). And I have noidea how the result will look like.

**Comment by anjew175** [comment]
Has been updated to adhere to requested changes

**Comment by ltem** [comment]
The code looks fine (Note: it won't spawn a moneycrate on death) but the artwork looks still glitchy on the desert terrain.

screens

**Comment by reaperrr** [comment]
Rebased and updated the sprite to work on all terrain:
truk-tem
truk-des

**Review by GraionDilach — APPROVED** [review]
:+1: then.

**Review by ltem — APPROVED** [review]
Looks good.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/49a06a019743b9d9ed11625fac07a5e15950592a

**Comment by anjew175** [comment]
Thank you for doing up the sprites @reaperrr 
Look great.

I had no idea sprites could look good on one terrain and shit on another

**Comment by reaperrr** [comment]
> I had no idea sprites could look good on one terrain and shit on another

Unlike RA, TD currently uses the terrain-specific palettes for units and buildings, too. That's fine for TD sprites because Westwood designed the sprites to not use tileset-specific colors, but when converting unit shps from RA to TD or vice-versa using the normal palettes, XCC Mixer simply uses the closest match it can find, or first perfect match, and quite often that's a terrain color index. I have some custom modified palettes here where these tileset colors are replaced with pink, so XCC Mixer won't use those indexes when converting to shp.

**Pull request merged**